lib/ok.ex

```diff
@@ -231,7 +251,21 @@ defmodule OK do
       end
     end
   end
+  defp nest([{:ok, _} = normal | []]) do
+    # last line of the with block is an ok literal
+    quote do
+      unquote(normal)
+    end
+  end
+  
+  defp nest([{{:., _, [{:__aliases__, _, [:OK]}, :success]}, _, _} = normal | []]) do
+    # last line of the with block is an OK.success macro line
+    quote do
+      unquote(normal)
+    end
+  end
```

Comment:
> Someone could write a function that always returned {:ok, value} and it would produce the same errors but not be handled by these clauses.

What you write here is ambiguous upon reflection: Do you mean that someone could write a function as the last line, that function always returning `{:ok, value}`? Or do you mean that someone could write a function with an `OK.with` block in it that always returns `{:ok, value}`? If the former, then that is exactly what this PR is for - to match on the special case where either an ok literal (`{:ok, value}`) is returned or if the `OK.success` macro is used. There are no other special cases to consider. I'll speak to each of your 3 examples:

```
# 1
{:error, reason}
```
This is also non-sensical and _should_ produce the warning. If the programmer writes this as their last statement _in the happy path_, then that means there is no happy path. If there is some weird exception to this reasoning, then that is completely an edge case not worth considering.

```
# 2
OK.failure(reason)
```
Same as above.

# 3
_result = {:ok, value}
```

This is also an edge case, albeit less fanciful. I have used labeled returns, although never inside of a `with` construct. Usually it's within an anonymous function (e.g. within an `Enum.reduce`) where it may not be obvious what exactly is returning. But this is often a hack and a code smell, as it indicates it should be broken out into its own function. So along the lines of encouraging good code, I would see a warning produced for this to be a good thing. If you really must have a labeled return for the `OK.with` statement, there are still two equivalent ways: 1) `_result = OK.with...` and 2) `{:ok, _result = value}`, neither of which produces the warning. But I personally don't see this as a legitimate concern.

> Is there some clever way we can use Macro.expand to take the cast ast and expanded it deleting unused paths. It should be possible to make use of the code which generates the warnings.

There may totally be a more clever way and ultimately more elegant way. I considered other options that analyze the incoming expression (although not with `Macro.expand`) but I thought the pattern matching is explicit, covers the use cases, and is obvious what it does.